Description: [PATCH] Add user_name to the job_submit_lua plugin
Author: Doug Jacobsen <dmjacobsen@lbl.gov>
Date: Wed Mar 7 10:14:43 2018 -0700
Last-Update: 2018-06-25

Bug 4549

diff --git a/src/plugins/job_submit/lua/job_submit_lua.c b/src/plugins/job_submit/lua/job_submit_lua.c
index 39052d2..4fa302a 100644
--- a/src/plugins/job_submit/lua/job_submit_lua.c
+++ b/src/plugins/job_submit/lua/job_submit_lua.c
@@ -57,6 +57,7 @@
 #include "src/common/slurm_xlator.h"
 #include "src/common/assoc_mgr.h"
 #include "src/common/xlua.h"
+#include "src/common/uid.h"
 #include "src/slurmctld/locks.h"
 #include "src/slurmctld/slurmctld.h"
 #include "src/slurmctld/reservation.h"
@@ -857,6 +858,10 @@ static int _get_job_req_field(const struct job_descriptor *job_desc,
 		lua_pushnumber (L, job_desc->time_min);
 	} else if (!xstrcmp(name, "user_id")) {
 		lua_pushnumber (L, job_desc->user_id);
+	} else if (!xstrcmp(name, "user_name")) {
+		char *username = uid_to_string_or_null(job_desc->user_id);
+		lua_pushstring (L, username);
+		xfree(username);
 	} else if (!xstrcmp(name, "wait4switch")) {
 		lua_pushnumber (L, job_desc->wait4switch);
 	} else if (!xstrcmp(name, "work_dir")) {
